### PR TITLE
Add a new Virtual Forwarding Instance (VFI) type to the openflow library

### PR DESCRIPTION
**Pull request by sdmodi**
Reviewer: @andi-bigswitch @ronaldchl 

We need a new VFI type for the VxLAN feature.

### CODE REVIEW

**Comment by bsn-abat** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.bigswitch.com/job/loxigen_pull_req/220/

**Comment by sdmodi** [comment]
@andi-bigswitch @ronaldchl can you please review this. This is holding up the changes for VxLAN.

File: java_gen/pre-written/src/main/java/org/projectfloodlight/openflow/types/VFI.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    @Override
+    public VFI applyMask(VFI mask) {
+        return VFI.ofVfi(this.vfi & mask.vfi);
+    }
+
+    @Override
+    public int compareTo(VFI o) {
+        return Shorts.compare(vfi, o.vfi);
```
Comment (andi-bigswitch) [inline java_gen/pre-written/src/main/java/org/projectfloodlight/openflow/types/VFI.java]: Note that short is signed, which will create potentially unexpected orderings (putting vfis with the highest bit set first).

For a more predictable ordering, you'll need to do `Integer.compare(vfi & 0xffff, o.vfi & 0xffff)`

Alternatively, you could just store the vfi as an integer internally, and `c.readUnsignedShort` and `c.writeUnsignedShort`.

**Comment by andi-bigswitch** [comment]
Otherwise LGTM

**Comment by ronaldchl** [comment]
LGTM other than what @andi-bigswitch has pointed out

File: java_gen/pre-written/src/main/java/org/projectfloodlight/openflow/types/VFI.java (hunk 1 above)
Comment (sdmodi) [inline java_gen/pre-written/src/main/java/org/projectfloodlight/openflow/types/VFI.java]: I'll do the former. Thanks.

**Comment by sdmodi** [comment]
SELF-ACK

**Comment by bsn-abat** [comment]
ABAT: START_MERGE:  Log file at http://vss1/abat/2016.02.19.1844-m.loxigen.master/abat.log

**Comment by bsn-abat** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.bigswitch.com/job/loxigen_pull_req/224/

**Comment by bsn-abat** [comment]
ABAT: ACCEPT:  Successfully merged

In case you want to see the build log, check out :
Log file at http://vss1/abat/2016.02.19.1844-m.loxigen.master/abat.log
/cc

**Pull request merged**